debezium-core/src/main/java/io/debezium/connector/LegacyV1AbstractSourceInfoStructMaker.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector;
+
+import java.util.Objects;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+
+/**
+ * Legacy source info that does not enforce presence of the version and connector fields
+ *
+ * @author Jiri Pechanec
+ */
+public abstract class LegacyV1AbstractSourceInfoStructMaker<T extends AbstractSourceInfo> implements SourceInfoStructMaker<T> {
+    public static final String DEBEZIUM_VERSION_KEY = "version";
+    public static final String DEBEZIUM_CONNECTOR_KEY = "connector";
+ 
+    private String version;
+    private String connector;
+    protected String serverName;
+
+    public void init(String connector, String version, String serverName) {
```

Comment:
Would be nice to have those as constructor parameters, allowing for final field assignment, tieing in with the comment above about not exposing the classes publicly.

Reply:
@gunnarmorling Done